Iterate error page guidance

davidhunter08 opened the pull request:
## Description

Consolidating the 'problem with service', 'problem logging in', 'get urgent help' and 'contacting the NHS App' into one 'Error and status pages' guidance.

[Similar to this ONS guidance.](https://service-manual.ons.gov.uk/design-system/patterns/error-status-pages)

## Related to
- https://github.com/nhsuk/nhsapp-frontend/issues/263
- https://github.com/nhsuk/nhsapp-frontend/issues/119

## Draft guidance

<details>
<summary>V1</summary>

![image](https://github.com/user-attachments/assets/05184423-09d6-4130-8cc2-55895a035c9d)

</details>

<details>
<summary>V2</summary>

![image](https://github.com/user-attachments/assets/87a98745-5d0e-4b62-a445-df909d958074)

</details>

<details>
<summary>V3</summary>

![image](https://github.com/user-attachments/assets/b87739f8-5777-44c9-9754-8c6dd82ed6d4)

</details>

<details>
<summary>V4</summary>

![image](https://github.com/user-attachments/assets/54fda9a5-64fa-44bd-bfb8-8c6a93442cf8)

</details>

<details>
<summary>V5</summary>

![image](https://github.com/user-attachments/assets/51c9a2d9-dc15-4658-a4ab-c58728038a9d)

</details>

<details>
<summary>V6</summary>

![image](https://github.com/user-attachments/assets/2bb4f1db-a401-4606-8289-6c3d30f48cb3)

</details>

<details>
<summary>V7</summary>

![image](https://github.com/user-attachments/assets/daf52b6e-2077-40a1-8f27-e0f21492858c)

</details>

<details>
<summary>V8 (Latest version)</summary>

![image](https://github.com/user-attachments/assets/a616ff36-4f14-4d72-9bfd-9d9ab690d610)

</details>

[Screenshots with feedback on Mural](https://app.mural.co/t/nhsdigital8118/m/nhsdigital8118/1737373079536/cc46abcaa0a9f6aeeb7427e92d9bd00c68b924c1?wid=112-1741195959451)

davidhunter08 commented:
## Draft guidance - V2

- expanded on 'error codes' guidance (but should the heading be "Error codes"?)

![image](https://github.com/user-attachments/assets/87a98745-5d0e-4b62-a445-df909d958074)

Graham-Pembrey commented:
This is looking great. Thoughts on iterations ahead of our chat tomorrow:

- Medical signposting: the clinical team have recommended including 111 on more or less all error pages, with the rationale that we can't be sure what users are trying to do and whether they might need help. Our current guidance makes some reasonable point though about when we might want to leave it out. Perhaps we need to strike a balance.
- Content designers have expressed that guidance on writing headings would be helpful: e.g. when to use 'You' or 'We', or phrase differently. This could be something that we work in.
- 'Other things you can do' vs 'Other options' as the alternatives H2? 
- 'You need to update the NHS App' example just needs a small tweak in the body content: 'You will not be able to ~update~**use** the NHS App...'
- Do we need to specify that CTAs should use a button rather than link text and why? Something that was questioned in feedback on the Mural board.

davidhunter08 commented:
Do we consider this to be an error/status page? A user cannot continue to the next page.

<img width="545" alt="Screenshot 2025-04-03 at 13 44 31" src="https://github.com/user-attachments/assets/8f334f5b-4217-4d8d-a0c9-4dbd0c37e40f" />

davidhunter08 commented:
Thoughts on 'buttons v links' usage on error and status pages:
- **Primary button** = this will solve your problem, eg: "Update the app"
- **Secondary button** = this might solve your problem, eg: "Try again"
- **Link** - going elsewhere might help you, eg: "Use 111 to request emergency prescription"

Graham-Pembrey commented:
**Rough ideas for headings guidance**

You can use the main heading to explain the problem in a neutral way. For example:

- No internet 
- No appointments available to book at this time
- Face ID not found
- There is a problem with the service

Only include "you" in headings when you're telling the user to take a direct action. For example:

- You need to update the NHS App
- Check you're registered with a GP in England

If the NHS App cannot show something for a technical reason, you can start the heading with "we" to make clear that it's not the user's fault. For example:

- We could not show your approved prescriptions
- We could not log you in

davidhunter08 commented:
# Back links and breadcrumbs on error pages - desk research

## Problems with including breadcrumbs or back links on error pages

Including breadcrumbs or back links on error pages (404, 500, 503) can create several problems:

1. **False sense of navigation** - Breadcrumbs suggest a valid location within the site hierarchy, but error pages exist outside the normal site structure. This can confuse users about where they actually are.

2. **Broken navigation paths** - If a user arrived at an error page through an external link or bookmark to a page that no longer exists, breadcrumbs would represent a path that isn't valid, potentially leading to more errors.

3. **Security concerns** - For server errors like 500 or 503, exposing site structure through breadcrumbs could potentially reveal sensitive information about the application architecture that shouldn't be public.

4. **Back link failures** - "Back" links on error pages can create navigation loops if the previous page was the source of the error, or if the error occurred during form submission.

5. **User confusion** - When users encounter an error, providing complex navigation elements can distract from the primary message about what went wrong and what to do next.

Instead of breadcrumbs or back links, error pages are typically more effective when they offer a clean, simple experience with:
- Clear explanation of the error
- A search box
- Link to the homepage
- Contact information for support
- Possibly some popular links or a simplified site map

## Scenarios where back links or breadcrumbs could be useful

Back links or breadcrumbs could be useful in certain error scenarios:

1. **Partial system errors** - When only a specific feature or section fails but the rest of the site is functional, breadcrumbs can help users navigate to working areas.

2. **Authenticated sections** - If a user encounters an error in a members-only area, breadcrumbs can help them navigate back to other authenticated sections without losing their session.

3. **Multi-step processes** - If an error occurs during a checkout or form submission process, breadcrumbs showing the steps can help users understand where they are and how to potentially restart from an earlier point.

4. **Expected temporary errors** - For 503 "Service Unavailable" pages during scheduled maintenance, breadcrumbs could show users which section they were trying to access, helping them return later.

5. **Authorised access errors (403)** - When users lack permission for a specific page but have access to parent sections, breadcrumbs can guide them back to areas they can use.

6. **Dynamic content errors** - When specific content is unavailable but its category/section exists, breadcrumbs can help users find related content.

7. **Search result errors** - If a user clicks a search result that no longer exists, breadcrumbs could show the search path they took to help refine their query.

The key is context-awareness - breadcrumbs or back links work best when they represent valid paths the user can actually follow, rather than paths that might lead to additional errors.

Graham-Pembrey commented:
Suggested 'Next steps' for the guidance:

> ### Next steps
> 
> We want to learn more about:
> 
> - "try again" buttons on errors, and how we can best help users when they fail to solve the problem  
> - how we can avoid users needing to manually input error codes, for example by pre-populating these codes on our contact form
> - how this guidance could evolve into separate pages covering specific errors, when we have enough evidence to do this

davidhunter08 commented:
## Draft guidance - V4

- added next steps

![image](https://github.com/user-attachments/assets/54fda9a5-64fa-44bd-bfb8-8c6a93442cf8)

davidhunter08 commented:
## Draft guidance - V5

Updated guidance based on feedback, including:
- changed title to 'Error pages'
- added links to GOV.UK design patterns - still to discuss need for 'Page not found' pattern
- restructured 'Writing headings' section
- added 'Using calls to action and links` section
- iterated medical signposting and error code sections
- update examples

<details>
<summary>Screenshot of guidance</summary>

![image](https://github.com/user-attachments/assets/a7d5232b-08b1-421c-bd12-53c4301abf12)

</details>

davidhunter08 commented:
## Draft guidance - V6

Iterated guidance after content crit, including:
- moving examples to within the 'How to use' section
- adding a main example to the top of the page
- removing usage of primary button

<details>
<summary>Screenshot of guidance</summary>

![image](https://github.com/user-attachments/assets/2bb4f1db-a401-4606-8289-6c3d30f48cb3)

</details>

Graham-Pembrey commented on docs/examples/error-pages/confirmed-prescriptions.njk:
```diff
@@ -0,0 +1,29 @@
+
+    <h1 class="nhsuk-u-margin-bottom-5">We could not show your confirmed prescriptions</h1>
+
+    <p>Contact your GP or pharmacy directly if you need help with a precsription now.</p>
```
Minor typo: 'precsription' > 'prescription'.

Graham-Pembrey commented on docs/patterns/error-page.md:
```diff
@@ -0,0 +1,128 @@
+
+- blame the user
+- include breadcrumbs
+- display vague terms of jargon like "500", "504", "bad request" or "we are experiencing technical difficulties
```
Just needs the closing quote marks at the end of this line: 

> technical difficulties"

Graham-Pembrey commented on docs/examples/error-pages/check-internet-connection.njk:
```diff
@@ -0,0 +1,27 @@
+
+    <h1 class="nhsuk-u-margin-bottom-5">Check your internet connection</h1>
+
+    <p>Try using WiFi or move to a place with better internet connection, then try again.</p>
```
Realised we had a content crit comment about the word "try" being repeated a few times in this example. Should we drop ", then try again." from the body content?

Graham-Pembrey reviewed (changes requested):
A few minor points mentioned in comments.

davidhunter08 commented:
## Draft guidance - V7

- fixed typos
- updated examples
- adding link to 'Page not found' pattern
- iterated 'Call to actions' section

<details>
<summary>Screenshot of guidance</summary>

![image](https://github.com/user-attachments/assets/3d84ad1d-4643-436a-a303-55d80cbf5697)

</details>

Graham-Pembrey reviewed (changes requested):
Looks good to me. Just suggesting one small tweak to an example.
